Azure/azure-cli
# [ARM] `az ts`: Fix the issue that export template specs with no linked templates failed

**Description**
Issue: #17759

az ts export failed to export template specs with no linked templates (previously known as artifacts). This PR removes this [customer reported bug](https://github.com/Azure/azure-cli/issues/17759) in the code.

In addition, the preview flag from the template specs cmdlets and the --template-specs parameter has been removed. [Template Specs is GA](https://techcommunity.microsoft.com/t5/azure-governance-and-management/arm-template-specs-now-ga/ba-p/2402618) and these flags are no longer necessary.


**Testing Guide**


1. Create a template spec with no linked_templates (also known as artifacts). 
2. Export the template spec  - az ts export -s {} --output-folder {}. The command should succeed.
      az ts export 
    -s  /subscriptions/{}/resourceGroups/{}/providers/Microsoft.Resources/templateSpecs/{}/versions/{} 
  --output- older ./exported

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.
